data_sources/dispatcher.py

```diff
@@ -0,0 +1,18 @@
+import exceptions
+from _cython_0_29_32 import generator
+
+
+class DataSourceDispatcher:
+    def __init__(self, *args) -> None:
+        self.__sources = {}
+        for source in args:
+            self.register_source(source)
+
+    def register_source(self, source) -> None:
+        self.__sources[source.get_name()] = source
```

Comment:
As type hints in Python is just a commentary, I'd add a check here that you're definitely adding an object which implements `DataSourceInterface` and throw an exception if not.